Remove unused main_thread leftover from cleanup rework.
https://github.com/aws/s2n-tls/pull/4871 stopped reading main_thread but left
the write; Clang on FreeBSD fails with -Werror,-Wunused-but-set-global.

--- a/utils/s2n_init.c
+++ b/utils/s2n_init.c
@@ -15,8 +15,6 @@
 
 #include "utils/s2n_init.h"
 
-#include <pthread.h>
-
 #include "api/unstable/cleanup.h"
 #include "crypto/s2n_fips.h"
 #include "crypto/s2n_libcrypto.h"
@@ -35,7 +33,6 @@
 
 static void s2n_cleanup_atexit(void);
 
-static pthread_t main_thread = 0;
 static bool initialized = false;
 static bool atexit_cleanup = false;
 int s2n_disable_atexit(void)
@@ -59,8 +56,6 @@
      */
     POSIX_ENSURE(!initialized, S2N_ERR_INITIALIZED);
 
-    main_thread = pthread_self();
-
     if (getenv("S2N_INTEG_TEST")) {
         POSIX_GUARD(s2n_in_integ_test_set(true));
     }
